### PR TITLE
NMS-16239: OpenShift: Document the impact of disabling allowPrivilegeEscalation

### PR DESCRIPTION
**Pull request by Bonrob2**
…Escalation

NMS-16239: Added information on issue around allowPrivilegeEscalation being set to false, and how to deal with it.

### All Contributors

* [ ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://opennms.atlassian.net/) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the Jira link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the Jira issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-16239

### CODE REVIEW

File: docs/modules/deployment/pages/core/containers.adoc [hunk 1]
```diff
@@ -58,14 +58,14 @@ For a list of values and their defaults, see https://github.com/OpenNMS/helm-cha
...
-* Removing `CAP_NET_RAW` capability or setting allowPrivilegeEscalation to false will impact ICMP and auto discovery features.
-  ** WORKAROUND: Use a Minion that is located outside of the OpenShift cluster.
+* If you set `allowPrivilegeEscalation` to false, you cannot use ICMP and the auto discovery feature from the {page-component-title} core.
+Make sure to remove `CAP_NET_RAW` from `allowedCapabilities` when you set `allowPrivilegeEscalation` to false.
 
-  * By default, OpenNMS will create a ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount.
-  The user used to install OpenNMS must have the required permissions to make these modifications the OpenShift Cluster, including admin access.
-  There are options to disable the creation of these elements, but if you do so, you will need to figure out a way to get the pods to work.
+* By default, OpenNMS will create a ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount.
+The user who installs OpenNMS must have the required permissions to make these modifications the OpenShift cluster, including admin access.
```
Comment (mershad-manesh) [inline docs/modules/deployment/pages/core/containers.adoc:67]: Should we say `these modifications to the OpenShift cluster`?

Comment (Bonrob2) [inline docs/modules/deployment/pages/core/containers.adoc:67]:
```suggestion
The user who installs OpenNMS must have the required permissions to make these modifications to the OpenShift cluster, including admin access.
```

Comment (Bonrob2) [reply docs/modules/deployment/pages/core/containers.adoc]: Yes. Thanks for catching that!

File: docs/modules/deployment/pages/core/containers.adoc [hunk 2]
```diff
@@ -58,14 +58,14 @@ For a list of values and their defaults, see https://github.com/OpenNMS/helm-cha
...
-  ** WORKAROUND: Use a Minion that is located outside of the OpenShift cluster.
+* If you set `allowPrivilegeEscalation` to false, you cannot use ICMP and the auto discovery feature from the {page-component-title} core.
+Make sure to remove `CAP_NET_RAW` from `allowedCapabilities` when you set `allowPrivilegeEscalation` to false.
 
-  * By default, OpenNMS will create a ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount.
-  The user used to install OpenNMS must have the required permissions to make these modifications the OpenShift Cluster, including admin access.
-  There are options to disable the creation of these elements, but if you do so, you will need to figure out a way to get the pods to work.
+* By default, OpenNMS will create a ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount.
+The user who installs OpenNMS must have the required permissions to make these modifications to the OpenShift cluster, including admin access.
+There are options to disable the creation of these elements, but if you disable them, you will need to determine a way to get the pods to work.
```
Comment (mershad-manesh) [inline docs/modules/deployment/pages/core/containers.adoc:68]: I missed this one, in my earlier review.

The user can disable allowPrivilegeEscalation,ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount; In doing so the Core will run in unprivileged mode (Same behaviour as CAP_NET_RAW being removed and/or allowPrivilegeEscalation being set to False).

Maybe we can say something like 

```
By default, OpenNMS will create a ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount.
The user who installs OpenNMS must have the required permissions to make these modifications to the OpenShift cluster, including admin access.

Disabling ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount along with setting allowPrivilegeEscalation to false will run the {page-component-title} core in unprivileged mode.

```

Comment (Bonrob2) [inline docs/modules/deployment/pages/core/containers.adoc:68]: Disabling ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount along with setting allowPrivilegeEscalation to false will run the {page-component-title} core in unprivileged mode.

Comment (Bonrob2) [inline docs/modules/deployment/pages/core/containers.adoc:68]:
```suggestion
Disabling ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount along with setting allowPrivilegeEscalation to false will run the {page-component-title} core in unprivileged mode.
```

File: docs/modules/deployment/pages/core/containers.adoc [hunk 3]
```diff
@@ -58,14 +58,14 @@ For a list of values and their defaults, see https://github.com/OpenNMS/helm-cha
...
+* If you set `allowPrivilegeEscalation` to false, you cannot use ICMP and the auto discovery feature from the {page-component-title} core.
+Make sure to remove `CAP_NET_RAW` from `allowedCapabilities` when you set `allowPrivilegeEscalation` to false.
 
-  * By default, OpenNMS will create a ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount.
-  The user used to install OpenNMS must have the required permissions to make these modifications the OpenShift Cluster, including admin access.
-  There are options to disable the creation of these elements, but if you do so, you will need to figure out a way to get the pods to work.
+* By default, OpenNMS will create a ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount.
+The user who installs OpenNMS must have the required permissions to make these modifications to the OpenShift cluster, including admin access.
+Disabling ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount along with setting allowPrivilegeEscalation to false will run the {page-component-title} core in unprivileged mode.
```
Comment (Bonrob2) [inline docs/modules/deployment/pages/core/containers.adoc:68]:
```suggestion
Disabling ClusterRole, ClusterRoleBinding, Route, SecurityContextConstraints, and ServiceAccount along with setting `allowPrivilegeEscalation` to false will run the {page-component-title} core in unprivileged mode.
```

**Review by mershad-manesh — APPROVED** [review]

**Pull request merged**